src/gui/FileBrowser.cpp

```diff
@@ -362,59 +364,155 @@ QList<QString> FileBrowserTreeWidget::expandedDirs( QTreeWidgetItem * item ) con
+		bool songEditor = !(ke->modifiers() & Qt::ControlModifier);
+		// If shift is held, we send the item to a new sample track...
+		bool sampleTrack = ke->modifiers() & Qt::ShiftModifier;
+		// ...but only in the song editor. So, ctrl+shift enter does nothing
+		if (sampleTrack && songEditor){ openInNewSampleTrack(file); }
+		// Otherwise we send the item as a new instrument track
+		else if (!sampleTrack){ openInNewInstrumentTrack(file, songEditor); }
+	}
+
+	// When space is pressed, start a preview of the selected item
+	if (preview){ previewFileItem(file); }
+}
+
+
+
+
+void FileBrowserTreeWidget::keyReleaseEvent(QKeyEvent* ke)
+{
+	// Cancel previews when the space key is released
+	if (ke->key() == Qt::Key_Space && !ke->isAutoRepeat()){ stopPreview(); }
+}
+
+
+
+
+void FileBrowserTreeWidget::hideEvent(QHideEvent* he)
+{
+	// Cancel previews when the user switches tabs or hides the sidebar
+	stopPreview();
+	QTreeWidget::hideEvent(he);
+}
+
+
+
+
+void FileBrowserTreeWidget::contextMenuEvent(QContextMenuEvent * e )
+{
+	FileItem * file = dynamic_cast<FileItem *>( itemAt( e->pos() ) );
+	if( file != nullptr && file->isTrack() )
+	{
+		QMenu contextMenu( this );
+
+		contextMenu.addAction(
+			tr( "Send to active instrument-track" ),
+			[=]{ sendToActiveInstrumentTrack(file); }
+		);
 
 		contextMenu.addSeparator();
 
-		contextMenu.addAction(QIcon(embed::getIconPixmap("folder")),
-					tr("Open containing folder"),
-					this,
-					SLOT(openContainingFolder()));
+		contextMenu.addAction(
+			QIcon(embed::getIconPixmap("folder")),
+			tr("Open containing folder"),
+			[=]{ openContainingFolder(file); }
+		);
+
+		QAction* songEditorHeader = new QAction( tr("Song Editor"), nullptr );
+		songEditorHeader->setDisabled(true);
+		contextMenu.addAction( songEditorHeader );
+		contextMenu.addActions( getContextActions(file, true) );
 
-		contextMenu.exec(e->globalPos());
+		QAction* bbEditorHeader = new QAction( tr("BB Editor"), nullptr );
+		bbEditorHeader->setDisabled(true);
+		contextMenu.addAction( bbEditorHeader );
+		contextMenu.addActions( getContextActions(file, false) );
 
-		// The context menu has been executed so we can reset this member back to nullptr.
-		m_contextMenuItem = nullptr;
+		// We should only show the menu if it contains items
+		if (!contextMenu.isEmpty()){ contextMenu.exec( e->globalPos() ); }
 	}
 }
 
 
 
 
-void FileBrowserTreeWidget::mousePressEvent(QMouseEvent * me )
+QList<QAction*> FileBrowserTreeWidget::getContextActions(FileItem* file, bool songEditor)
 {
-	QTreeWidget::mousePressEvent( me );
-	if( me->button() != Qt::LeftButton )
-	{
-		return;
+	QList<QAction*> result = QList<QAction*>();
+	const bool fileIsSample = file->type() == FileItem::SampleFile;
+
+	QString destination = fileIsSample ? "AFP Instance" : "Instrument Track";
+	QString shortcutMod = songEditor ? "" : "Ctrl +";
+
+	QAction* toInstrument = new QAction(
+		tr("Send to new %1 (%2 Enter)")
+		.arg(destination, shortcutMod),
+		nullptr
+	);
```

Comment:
A couple of things here:
* I think "AFP" would be clearer written out in full as "AudioFileProcessor" - some users may not be familiar with the acronym.
* The substitutions here may not work in every language (e.g. "new" may need to inflect differently for each destination; "to" may be an inflection of the noun itself, not a separate word; etc.). I think the the destination should be included directly in the string to translate, rather than substituted in later.
* "Ctrl" is "⌘" on Mac - use the `UI_CTRL_KEY` macro instead. (Also, ideally "Ctrl" should be translated too, but we don't do that anywhere else, so feel free not to.)

Reply:
Should be satisfactory as of latest commit, unresolve and let me know if I missed anything.